scripts/bump.sh

```diff
@@ -1,56 +1,97 @@
 #!/bin/sh
 
-# ----------- Basic Checking -----------
-
-# Check that 1 arg has been supplied
-if [ "$#" != "1" ]; then
-	echo "[-] Exactly one argument is needed."
-	echo '[?] Example: `./bump.sh 0.2.6`'
-	exit 1
-fi
-
 dependencies=(
 	semver
 	git
 	npm
 )
 
-# Check that required binaries are installed
-for package in ${dependencies[@]}; do
-	if [ ! $(which $package) ]; then
-		echo "[-] All of '${dependencies[@]}' are needed to bump version"
+# ----------- Basic Checking -----------
+
+# check that current directory is repo root
+check_pwd () {
+	root="$(dirname $(realpath $(dirname "$0")))"
+	[ "$root" != "$(pwd)" ] && {
+		echo "[!] the script should be run from the root of the repo"
+		echo "[-] expected '$root'"
+		echo "[-] found '$(pwd)'"
+		exit 1
+	}
+	[ -z "$DRY_RUN" ] || echo "[i] Current directory is repo's root OK"
+}
+
+# Check that 1 arg has been supplied
+check_arg () {
+	if [ "${NB_ARGS}" != "1" ]; then
+		echo "[-] Exactly one argument is needed, got ${NB_ARGS}."
+		echo '[?] Example: `./bump.sh 0.2.6`'
 		exit 1
 	fi
-done
+	[ -z "$DRY_RUN" ] || echo "[i] Only argument has been given OK"
+}
+
+# Check that required binaries are installed
+check_dependencies () {
+	for package in ${dependencies[@]}; do
+		if [ ! $(which $package) ]; then
+			echo "[-] All of '${dependencies[@]}' are needed to bump version"
+			exit 1
+		fi
+	done
+	[ -z "$DRY_RUN" ] || echo "[i] Dependencies: '${dependencies[@]}' are installed on the system OK"
+}
 
 # Check that supplied version is semantically correct
-if [ ! "$1" = "$(semver $1)" ]; then
-	echo "[-] $1 is not a valid version number according to semver"
-	exit 1
-fi
+check_version_semantics () {
+	if [ ! "${NEW_VERSION}" = "$(semver ${NEW_VERSION})" ]; then
```

Comment:
maybe directly `"${NEW_VERSION}" != "$(semver ${NEW_VERSION})"`? :yum:

Reply:
You're right that's better!

Reply:
:+1: